### PR TITLE
Improve luminex exponent parser

### PR DESCRIPTION
**Pull request by alejandro-salgado**
add luminex mark to tests
change hardcoded number of header columns with dynamic reading
set quotes as optional in regex for luminex parsing

### CODE REVIEW

File: tests/parsers/luminex_xponent/luminex_xponent_structure_test.py [hunk 1]
```diff
@@ -89,6 +90,7 @@ def test_create_header() -> None:
     )
 
 
+@pytest.mark.luminex
```
Comment (slopez-b) [inline tests/parsers/luminex_xponent/luminex_xponent_structure_test.py:93]: Please add this mark to the `tool.pytest.ini_options` `markers` list in the `project.toml` so it does not complain when running the tests.

**Review by slopez-b — CHANGES_REQUESTED** [review]
LGTM, just a minimal change requested

File: tests/parsers/luminex_xponent/luminex_xponent_structure_test.py (hunk 1 above)
Comment (nathan-stender) [reply tests/parsers/luminex_xponent/luminex_xponent_structure_test.py:93]: Note: I'm not really opposed to adding these marks, however, given that you can already run tests just one parser by using `hatch run test tests/parsers/<parser>`, I wonder what the use case for this is that specifying the folder does not solve?

Comment (alejandro-salgado) [reply tests/parsers/luminex_xponent/luminex_xponent_structure_test.py]: With this way of running the tests, there is no need for the pytest mark

**Pull request merged**